addons/website_event_track/static/src/js/website_event_track_proposal.js

```diff
@@ -0,0 +1,182 @@
+odoo.define('website_event_track.website_event_track_proposal', function (require) {
+    'use strict';
+
+    var core = require('web.core');
+    var publicWidget = require('web.public.widget');
+    var utils=require('web.utils');
+
+    var _t = core._t;
+
+    publicWidget.registry.websiteEventProposalForm = publicWidget.Widget.extend({
+
+    selector: '.o_website_event_track_proposal_form',
+    events: {
+        'input input[name="partner_name"]': '_onNameInputPropagateOnContact',
+        'click .o_wetrack_add_contact_information': '_onToggleContactInformationClick',
+        'click .o_wetrack_proposal_submit_button': '_send',
+    },
+
+    /**
+     * Check validity of input fields : empty required fields or bad formatted ones are 
+     * marked as is-invalid and o_has_error.
+     * 
+     * @private
+     */
+    _check_error_fields: function () {
+        var form_valid = true;
+
+        this.$('.form-group').each(function (k, field) { 
+            var $field = $(field);
+
+            // Validate inputs for this field, except if it is a select2 field.
+            var inputs = $field.find('.form-control').not('.o_wetrack_select2_tags');
+            var invalid_inputs = inputs.toArray().filter(function (input, k) {
+                return !input.checkValidity();
+            });
+            
+            // Add error display and classes to invalid fields.
+            $field.removeClass('o_has_error').find('.form-control').removeClass('is-invalid');
```

Comment:
is-invalid is a bootstrap thingy or odoo thingy ? Are we using bootstrap 5 ?

Reply:
It is a bootstrap thing but of what I saw it is in BS 4 already. It is used in website_form for main website form snippet, so it should be ok :)